### PR TITLE
Extract sails

### PR DESCRIPTION
**Pull request by jmhooper**
This PR has the code for extracting sails. At the moment it builds off of the PR that extracts the sails config, and has some code to get unit tests green and to get authentication working.

At this point, the remaining work involves getting the rest of the request specs green.

### CODE REVIEW

**Comment by jmhooper** [comment]
The tests for this branch are passing, and the app appears to be working. There are still some TODOs, but due to the size of this branch, I'll move those to another branch.

This PR should be held until PRs that do the following are ready

- [ ] Quit using global objects previously provided by Sails
- [ ] Setup proper request and error logging

**Comment by jmhooper** [comment]
I removed the `HOLD` on this since it is going into `jmhooper-extract-sails-config` instead of `staging`, and `jmhooper-extract-sails-config` is held.

**Comment by brandocalrissian** [comment]
AHA! Will look at it.

File: config/log.js [hunk 1]
```diff
@@ -1,29 +1,3 @@
...
-  *                                                                          *
-  * You may also set the level to "silent" to suppress all logs.             *
-  *                                                                          *
-  ***************************************************************************/
-
-  level: process.env.SAILS_LOG_LEVEL || 'info'
-
-};
+module.exports = {
+  level: process.env.SAILS_LOG_LEVEL || 'info',
```
Comment (brandocalrissian) [inline config/log.js]: Does SAILS_LOG_LEVEL have any meaning anymore after this PR?

Comment (jmhooper) [inline config/log.js]: We still use it, but we should rename it to just `LOG_LEVEL`.

Comment (brandocalrissian) [inline config/log.js]: Ah, yes, in the manifest, right where it belongs.

Comment (brandocalrissian) [inline config/log.js]: Agreed!

**Comment by jmhooper** [comment]
Renamed the env var. See now that this is approved. It's a big PR so I'm gonna do a manual merge.

**Pull request closed without merging**